File: retroravelry/src/main/java/com/caseykulm/retroravelry/models/photo/Photo.java
package com.caseykulm.retroravelry.models.photo;

import com.google.auto.value.AutoValue;

@AutoValue public abstract class Photo {

  public abstract String id();
  public abstract String medium_url();
  public abstract String small_url();
  public abstract String square_url();
  public abstract String thumbnail_url();
  public abstract String sort_order();
  public abstract int x_offset();
  public abstract int y_offset();

}

File: retroravelry/src/main/java/com/caseykulm/retroravelry/models/response/photo/Photo.kt
package com.caseykulm.retroravelry.models.response.photo

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Photo(
    val id: String?,
    @Json(name = "medium_url") val mediumUrl: String?,
    @Json(name = "small_url") val smallUrl: String?,
    @Json(name = "square_url") val squareUrl: String?,
    @Json(name = "thumbnail_url") val thumbnailUrl: String?,
    @Json(name = "sort_order") val sortOrder: String?,
    @Json(name = "x_offset") val xOffset: Int,
    @Json(name = "y_offset") val yOffset: Int
)
